lab/lablib/midi/MidiFile.js

```diff
@@ -165,18 +220,43 @@ export class MIDIMetaEvent {
   /**
    * Constructor
    * @param {number} meta_type MIDIMetaType
-   * @param {Uint8Array} payload Raw bytes of message
+   * @param {Uint8Array} data Raw bytes of message
    */
-  constructor(meta_type, payload) {
+  constructor(meta_type, data) {
     this.meta_type = meta_type;
-    this.payload = payload;
+    this.data = data;
   }
 
   /**
    * @type {number[]}
    */
   get sort_key() {
-    return [MIDIMetaEvent.MAGIC, this.meta_type, ...this.payload];
+    return [MIDIMetaEvent.MAGIC, this.meta_type, ...this.data];
+  }
+
+  get byte_length() {
+    // 3 byte header + data
+    return 3 + this.data.length;
+  }
+
+  /**
+   * Encode the message to binary
+   * @param {DataView} data_view DataView to write to
+   * @param {number} offset Starting offset
+   * @return {number} New offset after writing
+   */
+  encode(data_view, offset) {
+    data_view.setUint8(offset, MIDIMetaEvent.MAGIC);
+    data_view.setUint8(offset + 1, this.meta_type);
+    data_view.setUint8(offset + 2, this.data.length);
```

Comment:
this is technically wrong, it should be a variable-length quantity.

This also means byte_length is wrong.

that said... the way I export scores isn't going to use variable-length messages.

Reply:
Leave it for now, if I ever add text labels, that might be the time to add it.